brokerapi/brokers/gcp_service_broker.go

```diff
@@ -561,6 +580,70 @@ func InitCatalogFromEnv() ([]models.Service, error) {
 
 	}
 
+	// set up bigtable custom plans
+	var btDynamicPlans map[string]BigtableDynamicPlan
+	btDynamicPlanJson := os.Getenv("BIGTABLE_CUSTOM_PLANS")
+
+	if btDynamicPlanJson != "" {
+		err = json.Unmarshal([]byte(btDynamicPlanJson), &btDynamicPlans)
+		if err != nil {
+			return []models.Service{}, fmt.Errorf("Error unmarshalling bigtable custom plan json %s", err)
+		}
+
+		// save cloudsql plans to database and construct mapping
+		for planName, planDetails := range btDynamicPlans {
+
+			exists, existingPlan, err := db_service.CheckAndGetPlan(planName, planDetails.ServiceId)
+
+			if err != nil {
+				return []models.Service{}, err
+			}
+
+			id, err := db_service.GetOrCreatePlanId(planName, planDetails.ServiceId)
+			if err != nil {
+				return []models.Service{}, err
+			}
+			currentPlanIds = append(currentPlanIds, id)
+
+			features := map[string]string{
```

Comment:
can you share this code with CloudSql? seems like building this features map and the metadata are the only unique pieces.